modules/n1ql_fts/pages/index.adoc

```diff
@@ -0,0 +1,589 @@
+----
+
+
+TIP: The *cbimport* location might vary according to where you have installed Couchbase. On *Mac* it is typically at /Applications/Couchbase Server.app/Contents/Resources/couchbase-core/bin
+
+* Go to the *indexes* folder and run the following command to create the index we are going to use.
+
+----
+curl -XPUT -H "Content-type:application/json" http://<USERNAME>:<PASSWORD>@<IP_ADDRESS>:8094/api/index/movies_shingle -d @movies_shingle.json
+----
+
+Finally, go to Couchbase's Web Console and click on the *Search* menu. If everything is right you should see something similar to the following:
+
+image:001-index-created.png[Index Created Successfully]
+
+
+== Searching for Movies
+
+=== Building a basic search
+
+Let's start by creating a query which searches for *Star War*. Note that we are searching for *Star War* instead of *Star WarS*. The goal is to build a simple search that already has some tolerance for typos.
+
+Run the following query via web console:
+
+[source, SQL]
+----
+
+select m.original_title, SEARCH_SCORE() AS score  from movies m where search(m, 
+    {
+       "should": { 
+                   "disjuncts": [
+                        {"field":"original_title", "match": "Star War"}, 
+                        {"field":"original_title", "match": "Star War", "fuzziness":1}
+                    ]
+             
+       }
+    }
+) 
+order by score desc
+limit 20
+
+----
+
+It will probably take a long time to run. You can see why in the query plan:
+
+image:002-slow-query-plan.png[Slow Query Plan]
+
+If you pay attention to the plan above, you will notice that no index has been used even though we have an index already in place. 
+
+Open the index we created via *Search -> movies_shingle*, you should see something similar to the following:
+
+image:003-index.png[movies index]
+
+Note that this definition only index documents with the type *_class* equals to *com.cb.fts.sample.entities.Movie*, and that is why our index wasn't used by our first query. If we rerun the same query filtering the result by the document type, it will much faster than before:
+[source, SQL,]
+----
+
+select m.original_title, SEARCH_SCORE() AS score  from movies m where search(m, 
+    {
+       "should": { 
+                   "disjuncts": [
+                        {"field":"original_title", "match": "Star War"}, 
+                        {"field":"original_title", "match": "Star War", "fuzziness":1}
+                    ]
+             
+       }
+    }
+) 
+and m._class = "com.cb.fts.sample.entities.Movie" 
+order by score desc
+limit 20
+
+----
+
+The query plan now shows a step called *IndexFtsSearch* which means, as you might guess, that our index is now being used.
+
+image:004-index-being-used.png[Index used]
+
+Now, let's try to understand what is going on with our query:
+
+[source, JavaScript, linenums]
```

Comment:
You can use `[source,JSON]` for source code formatting in our docs — the language is identified when you hover over the code.